bsp/bsp.cpp

```diff
@@ -166,14 +166,22 @@ void ExecuteWakeProcess::execute() {
 
   pid_t child = fork();
   if (child == 0) {
-    // Redirect stdout to the pipe
-    for (int i = 0; i < PIPES; ++i) {
+    // Close the reading side of the pipe
+    for (int i = 0; i < PIPES; ++i)
       close(pipefds[i][0]);
-      // !!!
-      if (pipefds[i][1] != i+1) {
-        dup2(pipefds[i][1], i+1);
-        close(pipefds[i][1]);
-      }
+
+    // We need to dup2 pipefds[i][0,PIPES) to [1,PIPES]
```

Comment:
`// We need to dup2 pipefds[i][0,PIPES) to [1,PIPES]` what motivates this?

Reply:
File descriptor 1=stdout, 2=stderr. So you need to put the file descriptors there.
Wake additionally allows for descriptors 3, 4, and 5 to be called out as output for particular output streams.

The BSP leverages this to keep the log/info/warning/error streams distinct so it can route the messages on these descriptors to the appropriate severity in the GUI.

Reply:
(although, as you can tell by the new 'quiet' option, the GUI does not appear to do a very good job of this routing!)